### PR TITLE
fix clock initialization for a restart

### LINKED ISSUE #870: Wave model cannot restart if mediator flag is set to startup
**Describe the bug**
The wave model cannot start if wave information is not in the coupler.  The solution should be to set
start_type = startup in the nems.configure file, but when doing this, WW3 gets the wrong model time.

**To Reproduce**
set KEEP to true in rt.sh
Run the cpld_control_p8 and cpld_restart_p8 tests.
In the cpld_restart_p8 directory change
start_type = continue to
start_type = initial
and model will have an error with a time mismatch.

**Expected behavior**

The wave model should be initialing with the current time, not the forecast start time.

### PR DESCRIPTION
**Pull request by pjpegion**
# Pull Request Summary
This fixes issue #870 
## Description

Fixes the logic for a restart when there are not wave in the coupler and start_type needs to be set to initial.
Is a change of answers expected from this PR?

No changes to existing tests

Please also include the following information: 
* Add any suggestions for a reviewer 
* @DeniseWorthen @JessicaMeixner-NOAA @aliabdolali 
* Mention any labels that should be added:  _bug_, _documentation_, _enhancement_, _new feature_
* Are answer changes expected from this PR? Please describe the changes and the reason why in addition to which of the following labels would apply: _mod_def change_, _out_grd change_, _out_pnt change_, _restart file change_, _Regression test_ 
-->

### Check list  

- [ ] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [ ] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested?
run the cpled_control_p8 and cpled_restart_p8 tests, and change 
start_type = continue to
start_type = initial
and the model now runs

* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
No, and I don't think we need to add a test.
* Have the matrix regression tests been run (if yes, please note HPC and compiler)?
No
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.)
I have tested cpld_control_p8 and cpld_restart_p8 and they both pass.

### CODE REVIEW

**Comment by MatthewMasarik-NOAA** [comment]
Thank you for finding and fixing this issue @pjpegion.

Note that the Github builds failed due to the known issue in the CI workflow for `dev/ufs-weather-model`.

**Comment by BrianCurtis-NOAA** [comment]
Before this can enter the UFSWM commit queue it needs a code review.

**Comment by JessicaMeixner-NOAA** [comment]
> Before this can enter the UFSWM commit queue it needs a code review.

@BrianCurtis-NOAA This sounds really smart - but I thought the procedure before was to not approve these until tests at the ufs-weather-model tests have gone through, so this seems like a change so I wanted to confirm my understanding of this new paradigm. 


Following up to review: 
@pjpegion this PR looks good to me, but I know there's been lots of discussions about another update to this.  Do we want to cherry-pick those changes in from @DeniseWorthen to this PR or move forward with this PR as is?

**Comment by BrianCurtis-NOAA** [comment]
> @BrianCurtis-NOAA This sounds really smart - but I thought the procedure before was to not approve these until tests at the ufs-weather-model tests have gone through, so this seems like a change so I wanted to confirm my understanding of this new paradigm.

Yes, it seemed weird to me we would do all this testing without the actual CM's in the subcomponents reviewing the PR's. Also, some reviews from subcomponent PR's held up the commit queue, and we felt this change was beneficial to all. So this is a shift in our methodology. I see it as more control going to the subcomponents on how they want to run their PR structure and review process, and all we would need to look for from UFSWM would be review approvals and no changes from the most recent code review approval.

This is a work-in-motion, so if something works better for you, let me (us) know and we'll try to get it working.

**Comment by pjpegion** [comment]
@BrianCurtis-NOAA I am happy to hear this change in the order too. I always thought it was backwards to have code review after all of the tests, so if there was any changes due to code review, all of the tests would have to be re-run.  

Denise proposed a cleaner solution that I have in this PR, but I don't know if the solution has been coded up.

**Comment by pjpegion** [comment]
I pulled in Denise's  solution for the restart logic.  My tests pass on hera.

**Review by JessicaMeixner-NOAA — APPROVED** [review]

**Comment by JessicaMeixner-NOAA** [comment]
@BrianCurtis-NOAA These changes are okay from the wave modeling code management side and I think this can be added to the ufs-weather-model queue.  In the meantime,  I have also requested reviews from @alperaltuntas and @DeniseWorthen  for their awareness and for any feedback or reviews they'd like to give.

**Review by MatthewMasarik-NOAA — APPROVED** [review]

File: model/src/wav_comp_nuopc.F90 [hunk 1]
```diff
@@ -574,10 +576,24 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
...
+    call ESMF_ClockGet( clock, startTime=startTime, currTime=currTime, rc=rc)
+    TimeOffset = currTime - startTime
+    call ESMF_TimeIntervalGet(TimeOffset, h_r8=toff, rc=rc)
+    write(msgstring,'(a,g14.7)')'TimeOffset: CurrTime - StartTime = ',toff
+    call ESMF_LogWrite(trim(msgString), ESMF_LOGMSG_INFO)
     ! Initial run or restart run
     if ( runtype == "initial") then
       call ESMF_ClockGet( clock, startTime=esmfTime, rc=rc)
       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+#ifndef CESMCOUPLED
```
Comment (alperaltuntas) [inline model/src/wav_comp_nuopc.F90]: This should be corrected as `#ifndef W3_CESMCOUPLED`. Other than this, it all looks good to me.

**Review by alperaltuntas — CHANGES_REQUESTED** [review]

File: model/src/wav_comp_nuopc.F90 (hunk 1 above)
Comment (pjpegion) [reply model/src/wav_comp_nuopc.F90]: Good catch.

**Review by alperaltuntas — APPROVED** [review]

**Comment by MatthewMasarik-NOAA** [comment]
@pjpegion we discussed this PR briefly in UFS apps/comps meeting.  The code review from WW3 mgrs is complete, and approved.  I believe confirmation that no answers are changed is needed before being considered for the UFS commit queue.  When you're able, please run the rt's and attach the log files to confirm this.

**Comment by DeniseWorthen** [comment]
@MatthewMasarik-NOAA Actually This PR should not change baselines for UFS. What we need is for @pjpegion to attach the intel/gnu logs to the UWM PR to move it into the Q there.

**Comment by MatthewMasarik-NOAA** [comment]
> @MatthewMasarik-NOAA Actually This PR should not change baselines for UFS. What we need is for @pjpegion to attach the intel/gnu logs to the UWM PR to move it into the Q there.

@DeniseWorthen thank you for the clarification.

**Review by DeniseWorthen — APPROVED** [review]

**Pull request merged**